core/execution/TRTEngine.cpp

```diff
@@ -20,14 +23,33 @@ TRTEngine::TRTEngine(std::string serialized_engine)
         util::logging::get_logger().get_reportable_severity(),
         util::logging::get_logger().get_is_colored_output_on()) {
     std::string _name = "deserialized_trt";
-    new (this) TRTEngine(_name, serialized_engine);
+    new (this) TRTEngine(_name, serialized_engine, empty_string);
 }
 
-TRTEngine::TRTEngine(std::string mod_name, std::string serialized_engine)
+TRTEngine::TRTEngine(std::vector<std::string> serialized_info)
+    : logger(std::string("[] - "),
+        util::logging::get_logger().get_reportable_severity(),
+        util::logging::get_logger().get_is_colored_output_on()) {
+    std::string _name = "deserialized_trt";
+    std::string device_info = serialized_info[0];
+    std::string engine_info = serialized_info[1];
+
+    new (this) TRTEngine(_name, engine_info, device_info);
+}
+
+TRTEngine::TRTEngine(std::string mod_name, std::string serialized_engine, std::string serialized_device_info = empty_string)
     : logger(std::string("[") + mod_name + std::string("_engine] - "),
         util::logging::get_logger().get_reportable_severity(),
         util::logging::get_logger().get_is_colored_output_on()) {
 
+    // Deserialize device meta data if device_info is non-empty
+    if (!serialized_device_info.empty())
+    {
```

Comment:
Just a style thing, can you put the { on the same line?

Reply:
Fixed it